core/Pos/Util/Modifier.hs

```diff
@@ -161,6 +165,27 @@ delete
     => k -> MapModifier k v -> MapModifier k v
 delete k (MapModifier m) = MapModifier $ HM.insert k Nothing m
 
+data KeyState v
```

Comment:
This whole block is present [here](https://github.com/input-output-hk/cardano-sl/blob/master-release-candidate/core/Pos/Util/Modifier.hs#L168-L187). It has some differences, but just aesthetic I think (see `kiToMaybe` vs `toMaybe`). @pva701 , mind double checking?